src/main/java/org/broadinstitute/hellbender/utils/config/ConfigUtils.java

```diff
@@ -0,0 +1,332 @@
+
+            hasSetConfigFactoryVariableDefaults = true;
+        }
+    }
+
+    /**
+     * Get the configuration file name from the given arguments.
+     * Modifies the given arguments to remove both the configuration file specification string
+     * and the configuration file name from the args.
+     *
+     * NOTE: Does NOT validate that the resulting string is a valid configuration file.
+     *
+     * @param args Command-line arguments passed to this program.
+     * @param configFileOption The command-line option indicating that the config file is next
+     * @return The name of the configuration file for this program or {@code null}.
+     */
+    public static final String getConfigFilenameFromArgs( final ArrayList<String> args, final String configFileOption ) {
+
+        Utils.nonNull(args);
+        Utils.nonNull(configFileOption);
+
+        String configFileName = null;
+
+        for ( int i = 0 ; i < args.size() ; ++i ) {
+            if (args.get(i).compareTo(configFileOption) == 0) {
+
+                // Get rid of the command-line argument name:
+                args.remove(i);
+
+                if ( i < args.size() ) {
+
+                    // Get and remove the specified config file:
+                    configFileName = args.remove(i);
+                    break;
+                }
+                else {
+                    // Option was provided, but no file was specified.
+                    // We cannot work under these conditions:
+
+                    String message = "ERROR: Configuration file not given after config file option specified: " + configFileOption;
+                    System.err.println(message);
+                    throw new UserException.BadInput(message);
+                }
+            }
+        }
+
+        return configFileName;
+    }
+
+    /**
+     * Get the given key from a {@link GATKConfig} class.
+     * @param key The key in the {@link GATKConfig} to get.
+     * @param <T> The type of the key to obtain.
+     * @return The value for the specified {@code key}.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T getFromConfig(String key) {
+
+        Utils.nonNull(key);
+
+        // Get our configuration:
+        final GATKConfig config = ConfigCache.getOrCreate( GATKConfig.class );
+
+        try {
+            for (Method m : GATKConfig.class.getDeclaredMethods()) {
+                if (m.getName().equals(key)) {
+                    return (T) m.invoke(config);
+                }
+            }
+        }
+        catch (IllegalAccessException e) {
+            // We shouldn't need to worry about this one, but just in case:
+            logger.fatal("Could not access the given key from GATKConfig class: " + key);
+        }
+        catch (InvocationTargetException e) {
+            logger.fatal("Could not resolve the given key from GATKConfig class: " + key);
+        }
+
+        // This should never happen.
+        // If it does, throw an exception:
```

Comment:
Once you combine the catch blocks I would move this into the catch block, I think that would be clearer.  Then get rid of the log message I think, the exception will carry the information you need.  
Also, in general when we throw exceptions like this we tend to throw a GATKException of some sort instead of the more general RunTimeException